dbt/include/athena/macros/adapters/relation.sql

```diff
@@ -1,8 +1,34 @@
 {% macro athena__drop_relation(relation) -%}
+  {%- set native_drop = config.get('drop', default='False') | as_bool -%}
+  {%- set rel_type_object = adapter.get_table_type(db_name=relation.schema, table_name=relation.identifier) -%}
+  {%- set rel_type = none if rel_type_object == none else rel_type_object.value -%}
+  {%- set natively_droppable = rel_type == 'iceberg_table' or relation.type == 'view' -%}
+
+  {%- if native_drop and natively_droppable -%}
+    {%- do drop_relation_sql(relation) -%}
+  {%- else -%}
+    {%- do drop_relation_glue(relation) -%}
+  {%- endif -%}
+{% endmacro %}
+
+{% macro drop_relation_glue(relation) -%}
+  {%- do log('Dropping relation via Glue and S3 APIs') -%}
   {%- do adapter.clean_up_table(relation) -%}
   {%- do adapter.delete_from_glue_catalog(relation) -%}
 {% endmacro %}
 
+{% macro drop_relation_sql(relation) -%}
+
+  {%- do log('Dropping relation via SQL only') -%}
+  {% call statement('drop_relation', auto_begin=False) -%}
+    {%- if relation.type == 'view' -%}
+      drop {{ relation.type }} if exists {{ relation.render() }}
+    {%- else -%}
+      drop {{ relation.type }} if exists {{ relation.render_hive() }}
```

Comment:
how do you handle `DDL execution timed out` issue here ?

Reply:
I have raised an issue [here](https://github.com/dbt-athena/dbt-athena/issues/312) to look at that separately. I can add something in the docs around this flag to call out the potential issue. I'm not actually seeing timeouts generally in my real dbt project (maybe 30-40 Iceberg tables being dropped + recreated several times a day) but it seems clear it's an issue in AWS' implementation of Athena, not a dbt-athena issue.